Code/CMake/Externals/TINYXML2.cmake

```diff
@@ -31,12 +31,17 @@
 #-----------------------------------------------------------------------------
 # TINYXML2
 set(proj TINYXML2)
+if(LINUX)
+  set(proj tinyxml2)
+endif()
 if(SV_USE_${proj})
-
   # If using toplevel dir, TINYXML2_DIR to be the SV_TINYXML2_DIR set by the
   # simvascular_add_new_external macro
   if(SV_EXTERNALS_USE_TOPLEVEL_BIN_DIR)
-      set(${proj}_DIR ${SV_${proj}_DIR}/share/lib/cmake/tinyxml2 CACHE PATH "Force ${proj} dir to externals" FORCE)
```

Comment:
This is going to break on mac and linux unless we copy to the bin directory in a different way.

Reply:
It resolves with the current externals on Linux (but doesn't break anything since we don't currently actually link to tinyxml2).  You are right that my version has explicit paths:

# The installation prefix configured by this project.
set(_IMPORT_PREFIX "/usr/local/sv/ext/2018.05/release/bin/gnu/4.8/x64/release/tinyxml2-6.2.0")

How do you install relatively?

Reply:
https://github.com/SimVascular/SimVascular/blob/f03f4e2b5d8dea278168d1496fb2d4dc30f68c4c/Externals/CMake/TINYXML2.cmake#L81

Reply:
In Externals (Externals/CMake/HDF5.cmake), you specify:

-DCMAKE_INSTALL_PREFIX:STRING=${SV_EXTERNALS_${proj}_BIN_DIR}

Isn't this the wrong type for CMAKE_INSTALL_PREFIX?

You do the same thing for TINYXML2.cmake:

-DCMAKE_INSTALL_PREFIX:STRING=share

Isn't CMAKE_INSTALL_PREFIX a PATH type?  Does it have a different meaning if you specify as a STRING?

What is the default installation path that the relative path gets appended to?  On Windows, it would be to C:/Program Files/.... which I don't want to install into and copy from.

Reply:
I hadn't noticed that, but it seems to figure it out. I'm not actually sure what the difference between path and string types are. I think paths type let you tab-complete in the cmake gui, but other than that, they are the same?  It is /usr/local, which requires permissions on Mac and linux, and I don't want to use that.

Reply:
Are you sure it's not ignoring the value you are specifying?  I thought you could have variables of the same name as long as they were different types in CMake?

Reply:
Yep, it is installing in the location that I'm specifying, so it can't be ignoring them.

Reply:
I can't seem to find that you can have the same variable with different types in the docs, but I swear I've specified variables without types that were ignored by the cmake build.  It could be from the set documentation: 

"Both types can exist at the same time with the same name but different values. When ${FOO} is evaluated, CMake first looks for a normal variable ‘FOO’ in scope and uses it if set. If and only if no normal variable exists then it falls back to the cache variable ‘FOO’."

"Finally, whenever a cache variable is added or modified by a command, CMake also removes the normal variable of the same name from the current scope so that an immediately following evaluation of it will expose the newly cached value."

A string is fundamentally different since it can be anything while a path (I believe) needs to refer to a single path.

Reply:
I think that is talking about how cmake deals with cached/non-cached variables, but I could be wrong.